### PR TITLE
Fee redesign in school domain.

### LINKED ISSUE #5922: Payment Entry against School Fees
- Add Student party type

### LINKED ISSUE #9847: Education domain Fee module redesign
- [x] Fee Request
       - Can be created against program and batch or student group
       - Will create Fee records in the backend
- [x] Fee
      - Payment request will be created on submission of the Fee records (Fee will be reference 
         doctype)
      - Payment Entry will be done against the student as a party. 
- [x] Outstanding Balance Reports

### PR DESCRIPTION
**Pull request by manassolanki**
The newly updated Fee design workflow is as described below:

1. First of all, we have created the accounts head "Academic Receivable" as receivable type account and "Academic Fee" as income type account
school-8000-desk 1

2. Then we can set these accounts as default receivable account and default income account. 

3. Then we will create the Fee Structure which contains the fee breakup and accounting details before any fee collection.
http://imgur.com/a/CBszO

4. After creating the Fee Structure, we can start with the Fee creation. This will include the following steps.
    i. Create and submit the fee records for a student. After submitting this record, a GL entries will be created in the backend, one against Academic Receivable (Debit) and another against the Academic Fee(Credit) considering student as a party.
    ii. You can create the payment request from the fee record itself simply by clicking on the make Payment Request.
    iii. Payment Entry can be done against the fee records.
The following link contains a GIF of the given workflow.
http://imgur.com/a/9V1XS

Fixes https://github.com/frappe/erpnext/issues/5922
Fixes https://github.com/frappe/erpnext/issues/9847

### CODE REVIEW

**Comment by nabinhait** [comment]
Change the label of "Debit to" to "Academic Receivable Account" and "Credit To" to "Income Account"

screen shot 2017-08-02 at 2 42 13 pm

---

Change the  label to Total Amount per Student, now it is confusing.

screen shot 2017-08-02 at 2 44 12 pm

---

If no contact email, while creating payment request from Fee, it gives error

screen shot 2017-08-02 at 2 54 40 pm

---

While creating Payment from Fee, getting this error.

screen shot 2017-08-02 at 3 01 45 pm

File: erpnext/schools/doctype/fees/fees.py [hunk 1]
```diff
@@ -6,17 +6,68 @@
...
+			self.grand_total += d.amount
+		self.outstanding_amount = self.grand_total
+		self.grand_total_in_words = money_in_words(self.grand_total)
+
+	def on_submit(self):
+
+		self.make_gl_entries()
+
+		if self.send_payment_request:
+			pr = make_payment_request(dt="Fees", dn=self.name, recipient_id=self.contact_email,
```
Comment (nabinhait) [inline erpnext/schools/doctype/fees/fees.py]: It gives error if no contact_email

File: erpnext/schools/doctype/fees/fees.py [hunk 2]
```diff
@@ -6,17 +6,68 @@
...
+					submit_doc=True, use_dummy_message=True)
+			frappe.msgprint(_("Payment request {0} created").format(getlink("Payment Request", pr.name)))
+
+
+	def make_gl_entries(self):
+		if not self.grand_total:
+			return
+		student_gl_entries =  self.get_gl_dict({
+							"account": self.debit_to,
```
Comment (nabinhait) [inline erpnext/schools/doctype/fees/fees.py]: extra indentation?

File: erpnext/schools/doctype/fees/fees.py [hunk 3]
```diff
@@ -27,6 +78,64 @@ def get_fee_list(doctype, txt, filters, limit_start, limit_page_length=20, order
 			order by due_date asc limit {0} , {1}'''
 			.format(limit_start, limit_page_length), student, as_dict = True)
 
+
+@frappe.whitelist()
+def get_payment_entry(dt, dn, bank_account=None):
```
Comment (nabinhait) [inline erpnext/schools/doctype/fees/fees.py]: Can't it be commonfied with `https://github.com/frappe/erpnext/blob/develop/erpnext/accounts/doctype/payment_entry/payment_entry.py#L667`?

File: erpnext/schools/doctype/fees/fees.json [hunk 4]
```diff
@@ -81,19 +142,18 @@
    "bold": 0, 
    "collapsible": 0, 
    "columns": 0, 
-   "fieldname": "academic_term", 
-   "fieldtype": "Link", 
+   "fieldname": "include_payment",
```
Comment (nabinhait) [inline erpnext/schools/doctype/fees/fees.json]: What is the use of this?

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/schools/doctype/fees/fees.json (hunk 4 above)
Comment (manassolanki) [inline erpnext/schools/doctype/fees/fees.json]: This will be used to send the payment request if we are creating the fee records from the fee schedule.

**Comment by rmehta** [comment]
@manassolanki please rebase and fix.

**Comment by manassolanki** [comment]
- [ ] Field rename
- [ ] contact email
- [ ] send email input in schedule
- [ ] get query for student group
- [ ] Read only for child table

**Comment by nabinhait** [comment]
Merged manually from console

**Pull request closed without merging**